lib/pure/times.nim

```diff
@@ -744,6 +744,336 @@ proc format*(info: TimeInfo, f: string): string =
+        # two digit format
+        info.hour = value[j..j+1].parseInt()
+        j += 2
+    of "m":
+        if value[j+1] in Digits:
+            # two digit format
+            info.minute = value[j..j+1].parseInt()
+            j += 2
+        else:
+            info.minute = parseInt($value[j])
+            j += 1
+    of "mm":
+        # two digit format
+        info.minute = value[j..j+1].parseInt()
+        j += 2
+    of "M":
+        var month: int
+        if value[j+1] in Digits:
+            # two digit format
+            month = value[j..j+1].parseInt()
+            j += 2
+        else:
+            month = parseInt($value[j])
+            j += 1
+        info.month = getMonth(month)
+    of "MM":
+        var month = value[j..j+1].parseInt()
+        j += 2
+        info.month = getMonth(month)
+    of "MMM":
+        case value[j..j+2].toLower():
+        of "jan":
+            info.month =  mJan
+        of "feb":
+            info.month =  mFeb
+        of "mar":
+            info.month =  mMar
+        of "apr":
+            info.month =  mApr
+        of "may":
+            info.month =  mMay
+        of "jun":
+            info.month =  mJun
+        of "jul":
+            info.month =  mJul
+        of "aug":
+            info.month =  mAug
+        of "sep":
+            info.month =  mSep
+        of "oct":
+            info.month =  mOct
+        of "nov":
+            info.month =  mNov
+        of "dec":
+            info.month =  mDec
+        else:
+            raise newException(ValueError, "invalid month") 
+        j += 3
+    of "MMMM":
+        if value.match(peg"^i'january'.*"):
+            info.month =  mJan
+            j += 7
+        elif value.match(peg"^i'february'.*"):
+            info.month =  mFeb
+            j += 8
+        elif value.match(peg"^i'march'.*"):
+            info.month =  mMar
+            j += 5
+        elif value.match(peg"^i'april'.*"):
+            info.month =  mApr
+            j += 5
+        elif value.match(peg"^i'may'.*"):
+            info.month =  mMay
+            j += 3
+        elif value.match(peg"^i'june'.*"):
+            info.month =  mJun
+            j += 4
+        elif value.match(peg"^i'july'.*"):
+            info.month =  mJul
+            j += 4
+        elif value.match(peg"^i'august'.*"):
+            info.month =  mAug
+            j += 6
+        elif value.match(peg"^i'september'.*"):
+            info.month =  mSep
+            j += 9
+        elif value.match(peg"^i'october'.*"):
+            info.month =  mOct
+            j += 7
+        elif value.match(peg"^i'november'.*"):
+            info.month =  mNov
+            j += 8
+        elif value.match(peg"^i'december'.*"):
+            info.month =  mDec
+            j += 8
+        else:
+            raise newException(ValueError, "invalid month") 
+    of "s":
+        if value[j+1] in Digits:
+            # two digit format
+            info.second = value[j..j+1].parseInt()
```

Comment:
Use parseutils.parseInt instead, or special case it. Turning 2 digits into an int is not hard!